### PR TITLE
Bugfix/unfreeze

### LINKED ISSUE #250: Idempotent machineSet and machineDeployment unfreeze
## Issue
machineSet and machineDeployment unfreeze operation has to be made atomic/idempotent. We see issues with unfreezing here. 

## Solution
Unfreezing should make sure that both conditions and labels on both machineSet and machineDeployment has to be removed.

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #250 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Bugfix: Improvements while unfreezing machineSet/machineDeployment
```

### CODE REVIEW

File: pkg/controller/controller_suite_test.go [hunk 1]
```diff
@@ -51,20 +52,122 @@ var (
 	controllerKindMachine = v1alpha1.SchemeGroupVersion.WithKind("Machine")
 )
 
-func newMachineSetFromMachineDeployment(machineSetCount int, machineDeployment *v1alpha1.MachineDeployment, replicas int32, statusTemplate *v1alpha1.MachineSetStatus) *v1alpha1.MachineSet {
-	return newMachineSetsFromMachineDeployment(1, machineDeployment, replicas, statusTemplate)[0]
+func newMachineDeployment(
+	machineDeploymentCount int,
```
Comment (ggaurav10) [inline pkg/controller/controller_suite_test.go]: Why do we need this `machineDeploymentCount` here?

File: pkg/controller/controller_suite_test.go [hunk 2]
```diff
@@ -51,20 +52,122 @@ var (
...
+	labels map[string]string,
+) []*v1alpha1.MachineDeployment {
+
+	intStr1 := intstr.FromInt(1)
+	machineDeployments := make([]*v1alpha1.MachineDeployment, machineDeploymentCount)
+	for i := range machineDeployments {
+		machineDeployment := &v1alpha1.MachineDeployment{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "machine.sapcloud.io",
+				Kind:       "MachineSet",
```
Comment (ggaurav10) [inline pkg/controller/controller_suite_test.go]: s/MachineSet/MachineDeployment/

File: pkg/controller/controller_suite_test.go (hunk 1 above)
Comment (prashanth26) [reply pkg/controller/controller_suite_test.go]: You are right, we don't need it. Have fixed it here - https://github.com/gardener/machine-controller-manager/pull/251/commits/d99cfa1f836c9e50d8396493c4e20487a66612f0

File: pkg/controller/controller_suite_test.go (hunk 2 above)
Comment (prashanth26) [reply pkg/controller/controller_suite_test.go]: Have fixed this as well - https://github.com/gardener/machine-controller-manager/pull/251/commits/d99cfa1f836c9e50d8396493c4e20487a66612f0

**Pull request merged**